proof/refine/RISCV64/InvariantUpdates_H.thy

```diff
@@ -200,6 +187,58 @@ lemma valid_irq_node'_ksWorkUnitsCompleted[simp]:
   "valid_irq_node' w (ksWorkUnitsCompleted_update f s) = valid_irq_node' w s"
   by (simp add: valid_irq_node'_def)
 
+lemma valid_irq_node'_ksReprogramTimer[simp]:
+  "valid_irq_node' w (ksReprogramTimer_update f s) = valid_irq_node' w s"
+  by (simp add: valid_irq_node'_def)
+
+lemma valid_release_queue_ksWorkUnitsCompleted[simp]:
+  "valid_release_queue (ksWorkUnitsCompleted_update f s) = valid_release_queue s"
+  by (simp add: valid_release_queue_def)
+
+lemma valid_release_queue_ksReprogramTimer[simp]:
+  "valid_release_queue (ksReprogramTimer_update f s) = valid_release_queue s"
+  by (simp add: valid_release_queue_def)
+
+lemma valid_release_queue_ksDomainTime[simp]:
+  "valid_release_queue (ksDomainTime_update f s) = valid_release_queue s"
+  by (simp add: valid_release_queue_def)
+
+lemma valid_release_queue_ksCurDomain[simp]:
+  "valid_release_queue (ksCurDomain_update f s) = valid_release_queue s"
+  by (simp add: valid_release_queue_def)
+
+lemma valid_release_queue_ksDomScheduleIdx[simp]:
+  "valid_release_queue (ksDomScheduleIdx_update f s) = valid_release_queue s"
+  by (simp add: valid_release_queue_def)
+
+lemma valid_release_queue_ksSchedulerAction[simp]:
+  "valid_release_queue (ksSchedulerAction_update f s) = valid_release_queue s"
+  by (simp add: valid_release_queue_def)
+
+lemma valid_release_queue'_ksSchedulerAction[simp]:
+  "valid_release_queue' (ksSchedulerAction_update f s) = valid_release_queue' s"
+  by (simp add: valid_release_queue'_def)
+
+lemma valid_release_queue'_ksWorkUnitsCompleted[simp]:
+  "valid_release_queue' (ksWorkUnitsCompleted_update f s) = valid_release_queue' s"
+  by (simp add: valid_release_queue'_def)
+
+lemma valid_release_queue'_ksReprogramTimer[simp]:
+  "valid_release_queue' (ksReprogramTimer_update f s) = valid_release_queue' s"
+  by (simp add: valid_release_queue'_def)
+
+lemma valid_release_queue'_ksDomainTime[simp]:
+  "valid_release_queue' (ksDomainTime_update f s) = valid_release_queue' s"
+  by (simp add: valid_release_queue'_def)
+
+lemma valid_release_queue'_ksCurDomain[simp]:
+  "valid_release_queue' (ksCurDomain_update f s) = valid_release_queue' s"
+  by (simp add: valid_release_queue'_def)
+
+lemma valid_release_queue'_ksDomScheduleIdx[simp]:
+  "valid_release_queue' (ksDomScheduleIdx_update f s) = valid_release_queue' s"
+  by (simp add: valid_release_queue'_def)
```

Comment:
@corlewis : could crunch or something like it automate some of this? I also saw some `empty_fail` that  looked derivable, unless I'm mistaken (happens a lot) crunch has a mode that can deal with those?

Reply:
This does smell automatable. If not with crunch, then probably with an update locale like we have for some of the other updators. Or potentially the `add_upd_simps` command which helpfully has no documentation at all, but sounds like it should be useful here.

Reply:
https://sel4.atlassian.net/browse/VER-1415